\begin{quote}
Start a promise chain \end{quote}
\href{http://cryto.net/~joepie91/blog/2016/05/11/what-is-promise-try-and-why-does-it-matter/}{\texttt{ How is it useful?}}\hypertarget{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12186}{}\doxysection{Install}\label{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12186}

\begin{DoxyCode}{0}
\DoxyCodeLine{\$ npm install p-\/try}

\end{DoxyCode}
\hypertarget{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12187}{}\doxysection{Usage}\label{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12187}

\begin{DoxyCode}{0}
\DoxyCodeLine{const pTry = require('p-\/try');}
\DoxyCodeLine{}
\DoxyCodeLine{(async () => \{}
\DoxyCodeLine{    try \{}
\DoxyCodeLine{        const value = await pTry(() => \{}
\DoxyCodeLine{            return synchronousFunctionThatMightThrow();}
\DoxyCodeLine{        \});}
\DoxyCodeLine{        console.log(value);}
\DoxyCodeLine{    \} catch (error) \{}
\DoxyCodeLine{        console.error(error);}
\DoxyCodeLine{    \}}
\DoxyCodeLine{\})();}

\end{DoxyCode}
\hypertarget{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12188}{}\doxysection{API}\label{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12188}
\hypertarget{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12189}{}\doxysubsection{p\+Try(fn, ...\+arguments)}\label{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12189}
Returns a {\ttfamily Promise} resolved with the value of calling {\ttfamily fn(...\+arguments)}. If the function throws an error, the returned {\ttfamily Promise} will be rejected with that error.

Support for passing arguments on to the {\ttfamily fn} is provided in order to be able to avoid creating unnecessary closures. You probably don\textquotesingle{}t need this optimization unless you\textquotesingle{}re pushing a {\itshape lot} of functions.\hypertarget{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12190}{}\doxysubsubsection{fn}\label{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12190}
The function to run to start the promise chain.\hypertarget{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12191}{}\doxysubsubsection{arguments}\label{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12191}
Arguments to pass to {\ttfamily fn}.\hypertarget{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12192}{}\doxysection{Related}\label{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12192}

\begin{DoxyItemize}
\item \href{https://github.com/sindresorhus/p-finally}{\texttt{ p-\/finally}} -\/ {\ttfamily Promise\#finally()} ponyfill -\/ Invoked when the promise is settled regardless of outcome
\item \href{https://github.com/sindresorhus/promise-fun}{\texttt{ More…}}
\end{DoxyItemize}\hypertarget{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12193}{}\doxysection{License}\label{md__home_denis_Documents_Faks_master_RS2_TennisAssociation_TennisAssociation_Client_node_modules_p_try_readme_autotoc_md12193}
MIT © \href{https://sindresorhus.com}{\texttt{ Sindre Sorhus}} 